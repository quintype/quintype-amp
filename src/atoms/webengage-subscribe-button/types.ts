import { DefaultTheme } from "styled-components";

export interface ButtonTypes {
  theme?: DefaultTheme;
  on?: string;
}
