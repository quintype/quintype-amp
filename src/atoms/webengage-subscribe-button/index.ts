import { WebengageSubscribeButton } from "./webengage-subscribe-button";
export { WebengageSubscribeButton };
