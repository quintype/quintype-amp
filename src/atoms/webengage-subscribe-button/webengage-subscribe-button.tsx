import React from "react";
import styled from "styled-components";
import { Bell } from "../icons";
import { withTheme } from "styled-components";
import { ButtonTypes } from "./types";
import { Spacer } from "../spacer";
import get from "lodash.get";

const StyledButton = styled.button`
  border-radius: 5px;
  background-color: ${(props) => props.theme.color.primaryColor};
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const SubscribeMessage = styled.span`
  color: ${(props) => props.theme.color.white};
  font: ${(props) => {
    const fontFamily = props.theme.font.family.primary;
    const fontWeight = props.theme.font.weight.bold;
    const fontSize = props.theme.font.size.s;
    return `${fontWeight} ${fontSize} ${fontFamily}`;
  }};
`;

export const WebengageSubscribeButtonBase = ({ theme, on, text }: ButtonTypes) => {
  const bellIconColor = get(theme, ["color", "white"], "currentColor");
  return (
    <StyledButton on={on} aria-label="bell icon">
      <Bell color={bellIconColor} />
      <Spacer align="horizontal" token="m" />
      <SubscribeMessage>{text || "Subscribe to instant updates"}</SubscribeMessage>
    </StyledButton>
  );
};

export const WebengageSubscribeButton = withTheme(WebengageSubscribeButtonBase);
