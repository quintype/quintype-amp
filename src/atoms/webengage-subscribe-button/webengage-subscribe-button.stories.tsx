import * as React from "react";
import { storiesOf } from "@storybook/react";
import { WebengageSubscribeButton } from "./webengage-subscribe-button";
import { Layout } from "../layout";
import { textStory, config } from "../../__fixtures__";

storiesOf("WebengageSubscribeButton", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("Default", () => <WebengageSubscribeButton />)
  .add("With custom text", () => <WebengageSubscribeButton text="Enable push notifications" />);
