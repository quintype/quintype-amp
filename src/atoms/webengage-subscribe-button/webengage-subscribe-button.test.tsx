import React from "react";
import { WebengageSubscribeButtonBase, SubscribeMessage } from "./webengage-subscribe-button";
import { shallow } from "enzyme";

describe("WebengageSubscribeButton", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<WebengageSubscribeButtonBase />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should apply custom text", () => {
    const wrapper = shallow(<WebengageSubscribeButtonBase text="foo" />);
    expect(wrapper.find(SubscribeMessage).text()).toBe("foo");
  });
});
