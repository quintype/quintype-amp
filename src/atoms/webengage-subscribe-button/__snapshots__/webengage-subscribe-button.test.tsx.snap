// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebengageSubscribeButton should match snapshot 1`] = `
<styled.button
  aria-label="bell icon"
>
  <Component
    color="currentColor"
  />
  <Spacer
    align="horizontal"
    token="m"
  />
  <styled.span>
    Subscribe to instant updates
  </styled.span>
</styled.button>
`;
